scapy/contrib/isotp.py

```diff
@@ -661,42 +744,52 @@ def __init__(self,
             listen_only=listen_only
         )
 
-        self.ins = impl
-        self.outs = impl
+        # Cast for compatibility to functions from SuperSocket.
+        self.ins = cast(socket.socket, impl)
+        self.outs = cast(socket.socket, impl)
         self.impl = impl
-
+        self.basecls = basecls
         if basecls is None:
             warning('Provide a basecls ')
-        self.basecls = basecls
 
     def close(self):
+        # type: () -> None
         if not self.closed:
             self.impl.close()
-            self.outs = None
-            self.ins = None
-            SuperSocket.close(self)
+            self.closed = True
 
     def begin_send(self, p):
+        # type: (Packet) -> int
         """Begin the transmission of message p. This method returns after
         sending the first frame. If multiple frames are necessary to send the
         message, this socket will unable to send other messages until either
         the transmission of this frame succeeds or it fails."""
-        if hasattr(p, "sent_time"):
-            p.sent_time = time.time()
 
-        return self.outs.begin_send(bytes(p))
+        if not self.closed:
+            if hasattr(p, "sent_time"):
+                p.sent_time = time.time()
+            self.impl.begin_send(bytes(p))
+            return len(p)
+        else:
+            return 0
 
     def recv_raw(self, x=0xffff):
+        # type: (int) -> Tuple[Optional[Type[Packet]], Optional[bytes], Optional[float]]  # noqa: E501
```

Comment:
multi-line

Reply:
again, the return type is filling the line.